homeassistant/components/schlage/switch.py

```diff
@@ -0,0 +1,114 @@
+"""Platform for Schlage switch integration."""
+
+from __future__ import annotations
+
+from functools import partial
+from typing import Any
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SchlageDataUpdateCoordinator
+from .entity import SchlageEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+    for device_id in coordinator.data.locks:
+        entities.extend(
+            [
+                BeeperSwitch(coordinator=coordinator, device_id=device_id),
+                LockAndLeaveSwitch(coordinator=coordinator, device_id=device_id),
+            ]
+        )
+    async_add_entities(entities)
+
+
+class BeeperSwitch(SchlageEntity, SwitchEntity):
```

Comment:
Since you're already using EntityDescriptions and both classes look the same, except for the action they do and the way to see if it's turned on. Maybe you can create a generic switch class. A good example I found was in Intellifire: https://github.com/home-assistant/core/blob/dev/homeassistant/components/intellifire/switch.py

Reply:
Done. (Ironically, I think this is actually more code, but I guess it's easier to maintain)

Reply:
It's a discussion on it's own, but this way you indeed have improved maintainability. Less code duplication and whenever you would want to add another one, you're going to benefit from this change :)